plugins/atomicfu/atomicfu-compiler/src/org.jetbrains.kotlinx.atomicfu.compiler/extensions/AtomicfuTransformer.kt

```diff
@@ -98,6 +102,17 @@ class AtomicfuTransformer(private val context: IrPluginContext) {
     private inner class AtomicTransformer : IrElementTransformer<IrFunction?> {
 
         override fun visitFunction(declaration: IrFunction, data: IrFunction?): IrStatement {
+            // Erase messages added by the Trace object from the function body:
+            // val trace = Trace(size)
+            // Messages may be added via trace invocation:
+            // trace { "Doing something" }
+            // or via multi-append of arguments:
+            // trace.append(index, "CAS", value)
+            (declaration.body as? IrBlockBody)?.let { body ->
```

Comment:
What if we have code like this?

```kotlin
fun foo() {
  if (cond()) {
    trace { "Doing something" }
  }
}
```

Reply:
Thank you, I've neglected that 😕.
Now trace is erased from both `IrContainerExpression` and `IrBlockBody`